lib/puppet/util/log.rb

```diff
@@ -147,7 +147,12 @@ def Log.newdestination(dest)
       Puppet.log_exception(detail)
 
       # If this was our only destination, then add the console back in.
-      newdestination(:console) if @destinations.empty? and (dest != :console and dest != "console")
+      if destinations.empty? && (dest.to_s != "console" && dest.intern != :console)
```

Comment:
Sorry that my previous comment was unclear. You just need to test one of these (either compare strings or compare symbols, but not both).